bin/local-env/run-e2e-tests.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Wrapper for the command `wp-scripts test:e2e`.
+ *
+ * This allows for a greater flexibility on what specs and tests can be run based on a given environment.
+ */
+
+const path = require( 'path' );
+const { spawn } = require( 'child_process' );
+const semver = require( 'semver' );
+const { spawnScript } = require( '@wordpress/scripts/utils' );
+
+const composeFile = path.resolve( process.cwd(), 'bin', 'local-env', 'docker-compose.yml' );
+
+// Retrieve the current WordPress version using the Docker CLI container.
+const prc = spawn( 'docker-compose', [ '-f', composeFile, 'exec', '-T', '-u', 'xfs', 'cli', 'wp', 'core', 'version' ] );
+
+prc.stdout.setEncoding( 'utf8' );
+
+prc.stdout.on( 'data', ( data ) => {
+	const wpVersion = data.toString().trim();
+
+	// The first 2 args are not needed - 'node', and the script name.
+	const suppliedArgs = process.argv.slice( 2 );
+
+	const testsToIgnore = [];
+
+	if ( semver.gte( '5.3.0', semver.coerce( wpVersion ) ) ) {
+		// Ignore tests that are not to be run in WP >= 5.3.0.
+		testsToIgnore.push( 'AMP Settings Screen should not allow AMP Stories to be enabled when Gutenberg is not active' );
+	}
+
+	const testNamePatterns = testsToIgnore.map( ( testName ) => {
+		return `--testNamePattern='^(?!${ testName }).*$'`;
+	} );
+
+	const cmdArgs = [ ...suppliedArgs, ...testNamePatterns ];
+
+	// Run E2E tests.
+	spawnScript( 'test-e2e', cmdArgs );
+} );
```

Comment:
I'm very inexperienced with E2E testing, but instead of creating this wrapper is not something like this possible:

```diff
diff --git a/tests/e2e/specs/amp-options.js b/tests/e2e/specs/amp-options.js
index 237fea8f..b874a08a 100644
--- a/tests/e2e/specs/amp-options.js
+++ b/tests/e2e/specs/amp-options.js
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+const semver = require( 'semver' );
+
 /**
  * WordPress dependencies
  */
@@ -54,6 +59,10 @@ describe( 'AMP Settings Screen', () => {
 	} );
 
 	it( 'should not allow AMP Stories to be enabled when Gutenberg is not active', async () => {
+		if ( semver.gte( '5.3.0', semver.coerce( process.env.WP_VERSION ) ) ) {
+			return;
+		}
+
 		await deactivatePlugin( 'gutenberg' );
 
 		await visitAdminPage( 'admin.php', 'page=amp-options' );
```

Reply:
Or rather:

```diff
diff --git a/tests/e2e/specs/amp-options.js b/tests/e2e/specs/amp-options.js
index 237fea8f..eeb1ad2a 100644
--- a/tests/e2e/specs/amp-options.js
+++ b/tests/e2e/specs/amp-options.js
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+const semver = require( 'semver' );
+
 /**
  * WordPress dependencies
  */
@@ -53,7 +58,9 @@ describe( 'AMP Settings Screen', () => {
 		expect( await page.$eval( '#amp-settings', ( el ) => el.matches( ':invalid' ) ) ).toBe( true );
 	} );
 
+	if ( semver.lt( '5.3.0', semver.coerce( process.env.WP_VERSION ) ) ) {
 		it( 'should not allow AMP Stories to be enabled when Gutenberg is not active', async () => {
+
 			await deactivatePlugin( 'gutenberg' );
 
 			await visitAdminPage( 'admin.php', 'page=amp-options' );
@@ -64,6 +71,7 @@ describe( 'AMP Settings Screen', () => {
 
 			await activatePlugin( 'gutenberg' );
 		} );
+	}
 
 	it( 'should allow AMP Stories to be enabled when Gutenberg is active', async () => {
 		await visitAdminPage( 'admin.php', 'page=amp-options' );
```

Reply:
As far as I'm concerned the WP version isn't provided in the environment.